sdk/messaging/azservicebus/message.go

```diff
@@ -323,6 +316,10 @@ func newReceivedMessage(amqpMsg *amqp.Message) *ReceivedMessage {
 		State:          MessageStateActive,
 	}
 
+	if msg.rawAMQPMessage.Data != nil && len(msg.rawAMQPMessage.Data) == 1 {
```

Comment:
Nit the `nil` check can be removed as a `nil` slice has a length of zero.

Reply:
Indeed! Fixed.